app/src/Page/Bcc/Edit/DomainRoles.elm

```diff
@@ -0,0 +1,252 @@
+                    ]
+                ]
+        _ ->
+            Card.config [ Card.attrs [ class "mb-3", class "shadow" ], Card.align Text.alignXsCenter ]
+                |> Card.block []
+            [ Block.custom <| 
+                Button.button [ Button.primary, Button.onClick ShowCreateNew ] [ text "Add new domain role" ]
+            , Block.custom <| 
+                div [] [ Html.br [] [] ]
+            , Block.custom <| 
+                Button.button [ Button.primary, Button.onClick ShowChooseFrom ] [ text "Choose from pre-defined list" ]
+            ]
+
+viewSelectDomainRole : Html Msg
+viewSelectDomainRole =
+  let
+    roles =
+      [ ("Specification Model", "Produces a document describing a job/request that needs to be performed. Example: Advertising Campaign Builder")
+      , ("Execution Model", "Performs or tracks a job. Example: Advertising Campaign Engine")
+      , ("Audit Model", "Monitors the execution. Example: Advertising Campaign Analyser")
+      , ("Approver", "Receives requests and determines if they should progress to the next step of the process. Example: Fraud Check")
+      , ("Enforcer", "Ensures that other contexts carry out certain operations. Example: GDPR Context (ensures other contexts delete all of a user’s data)")
+      , ("Octopus Enforcer", "Ensures that multiple/all contexts in the system all comply with a standard rule. Example: GDPR Context (as above)")
+      , ("Interchanger", "Translates between multiple ubiquitous languages.")
+      , ("Gateway", "Sits at the edge of a system and manages inbound and/or outbound communication. Example: IoT Message Gateway")
+      , ("Gateway Interchange", "The combination of a gateway and an interchange.")
+      , ("Dogfood Context", "Simulates the customer experience of using the core bounded contexts. Example: Whitelabel music store")
+      , ("Bubble Context", "Sits in-front of legacy contexts providing a new, cleaner model while legacy contexts are being replaced.")
+      , ("Autonomous Bubble", "Bubble context which has its own data store and synchronises data asynchronously with the legacy contexts.")
+      , ("Brain Context (likely anti-pattern)", "Contains a large number of important rules and many other contexts depend on it. Example: rules engine containing all the domain rules")
+      , ("Funnel Context", "Receives documents from multiple upstream contexts and passes them to a single downstream context in a standard format (after applying its own rules).")
+      , ("Engagement Context", "Provides key features which attract users to keep using the product. Example: Free Financial Advice Context")
+      ]
+  in
+    Form.group []
+      [ Select.select [ Select.id "role_select", Select.onChange (\name -> SelectRole name (getRoleDescriptionWithName name roles)) ]
+          (List.map (\(name, description) -> Select.item [ value name ] [ text (name ++ " (" ++ description ++ ")")]) roles)
```

Comment:
I think the description is hard to read as part of the HTML `select`

We could present them to the user as a (longish) list & ask the user to click on a line (as e.g. a link) instead?
Or do you think that this would complicate the layout on the page if we display a long list?

Another alternative would be a 'autocomplete select' similar to the one used [here](https://github.com/Softwarepark/Contexture/blob/master/app/src/Page/Bcc/Edit/Dependencies.elm#L228)

Reply:
How about we use a 3rd party component to display multi-line select options, this one for example: https://package.elm-lang.org/packages/mercurymedia/elm-smart-select/latest/ ?

Reply:
Yes this makes sense - the one I mentioned is [this](https://package.elm-lang.org/packages/sporto/elm-select/3.2.0/) one.
Maybe take a quick look if it's feasible with this one before we pull in another dependency?